From f4b20b16b80564f5578cc9119cf746447c979ef0 Mon Sep 17 00:00:00 2001
From: Ryan Eatmon <reatmon@ti.com>
Date: Tue, 6 Sep 2022 11:17:55 -0500
Subject: [PATCH] cryptodev-module: Remove 5.11 build patch

Upstream oecore is now carrying the same patch and having this
patch causes build errors.  So remove it.

SPDX-FileCopyrightText: Texas Instruments (assumed)
SPDX-License-Identifier: MIT
Upstream-status: Backport
Signed-off-by: Ryan Eatmon <reatmon@ti.com>
Signed-off-by: Paul Barker <paul.barker@sancloud.com>
---
 .../cryptodev/cryptodev-module_%.bbappend     |  2 --
 .../0001-Fix-build-for-Linux-5.11-rc1.patch   | 36 -------------------
 2 files changed, 38 deletions(-)
 delete mode 100644 meta-arago-distro/recipes-kernel/cryptodev/files/0001-Fix-build-for-Linux-5.11-rc1.patch

diff --git a/meta-arago-distro/recipes-kernel/cryptodev/cryptodev-module_%.bbappend b/meta-arago-distro/recipes-kernel/cryptodev/cryptodev-module_%.bbappend
index c210cc4e2613..92140bcb7ff7 100644
--- a/meta-arago-distro/recipes-kernel/cryptodev/cryptodev-module_%.bbappend
+++ b/meta-arago-distro/recipes-kernel/cryptodev/cryptodev-module_%.bbappend
@@ -2,6 +2,4 @@ PR_append = ".arago1"
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
-SRC_URI += "file://0001-Fix-build-for-Linux-5.11-rc1.patch"
-
 KERNEL_MODULE_AUTOLOAD += "cryptodev"
diff --git a/meta-arago-distro/recipes-kernel/cryptodev/files/0001-Fix-build-for-Linux-5.11-rc1.patch b/meta-arago-distro/recipes-kernel/cryptodev/files/0001-Fix-build-for-Linux-5.11-rc1.patch
deleted file mode 100644
index df07b9ea251c..000000000000
--- a/meta-arago-distro/recipes-kernel/cryptodev/files/0001-Fix-build-for-Linux-5.11-rc1.patch
+++ /dev/null
@@ -1,36 +0,0 @@
-From 55c6315058fc0dd189ffd116f2cc27ba4fa84cb6 Mon Sep 17 00:00:00 2001
-From: Joan Bruguera <joanbrugueram@gmail.com>
-Date: Mon, 28 Dec 2020 01:41:31 +0100
-Subject: [PATCH] Fix build for Linux 5.11-rc1
-
-ksys_close was removed, as far as I can tell, close_fd replaces it.
-
-See also: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8760c909f54a82aaa6e76da19afe798a0c77c3c3
-          https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=1572bfdf21d4d50e51941498ffe0b56c2289f783
-
-Upstream-status: Backport [55c6315058fc0dd189ffd116f2cc27ba4fa84cb6]
-
-Signed-off-by: Nishanth Menon <nm@ti.com>
----
- ioctl.c | 4 +++-
- 1 file changed, 3 insertions(+), 1 deletion(-)
-
-diff --git a/ioctl.c b/ioctl.c
-index 3d33238021dc..95481d4ff8e2 100644
---- a/ioctl.c
-+++ b/ioctl.c
-@@ -871,8 +871,10 @@ cryptodev_ioctl(struct file *filp, unsigned int cmd, unsigned long arg_)
- 		if (unlikely(ret)) {
- #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0))
- 			sys_close(fd);
--#else
-+#elif (LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0))
- 			ksys_close(fd);
-+#else
-+			close_fd(fd);
- #endif
- 			return ret;
- 		}
--- 
-2.31.0
-
-- 
2.34.1

